cpp/include/algorithms.hpp

```diff
@@ -587,6 +595,8 @@ void sssp(GraphCSRView<VT, ET, WT> const &graph,
  *
  * @param[in] directed               Treat the input graph as directed
  *
+ * @param[in] mg_batch               If set to true use SG BFS path when comms are initialized.
```

Comment:
This may go outside the scope of this PR, but I think we may better look for graph (rather than handle) to decide between SG & MG. If the graph is for single-GPU (or is not partitioned), we go for single GPU and go for multi-GPU otherwise.

With the current approach, each algorithm needs to consider three cases (SG, MG batch, MG). If the graph type dictates execution mode, each algorithm needs to consider only SG & MG (The caller needs to path SG graph for MG Batch).